### ISSUE TITLE
Running test_lint locally when lockfiles exist

### ISSUE THREAD

**Issue by wyardley** [GoogleCloudPlatform/cloud-foundation-toolkit#2622]
### TL;DR
The validation portion of `test_lint` can fail when there is a lockfile present locally, since most projects have the lockfiles .gitignored, but generation of them [can't be suppressed](https://github.com/hashicorp/terraform/issues/31533)

However, since in some places, the lockfile _is_ checked in, @apeabody says it probably isn't ideal to just force `-upgrade` on all `terraform init` calls as I'd originally proposed in the linked issue.

So perhaps there should be a way to control the behavior. Removing the lockfile might also be an option, but that seems riskier / more unexpected or undesirable from the standpoint of the end user.

Another option would be to do the whole init / validate in a temporary copy (perhaps the one that I think is already checked out?)

### Expected behavior

The validation to succeed when a new version has come out since the lockfile was last created.

### Observed behavior
Don't have the exact error handy now, but it's the one where the version is locked and can't be updated without `init -upgrade`

### Terraform Configuration

```hcl
n/a
```

### Terraform Version

```sh
Terraform v1.9.5
on darwin_arm64
```

### Additional information

See discussion starting at https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/2617#discussion_r1775481589
